tools/convert/converters/converter-bin.hpp

```diff
@@ -71,49 +71,56 @@ namespace rs2 {
                     return "BIN converter";
                 }
 
-                void convert(rs2::frameset& frameset) override
+                void convert(rs2::frame& frame) override
                 {
                     start_worker(
-                        [this, &frameset] {
-                            for (size_t i = 0; i < frameset.size(); i++) {
-                                rs2::depth_frame frame = frameset[i].as<rs2::depth_frame>();
+                        [this, &frame] {
+                        rs2::depth_frame depthframe = frame.as<rs2::depth_frame>();
 
-                                if (frame && (_streamType == rs2_stream::RS2_STREAM_ANY || frame.get_profile().stream_type() == _streamType)) {
-                                    if (frames_map_get_and_set(frame.get_profile().stream_type(), frame.get_frame_number())) {
-                                        continue;
-                                    }
+                        if (depthframe && (_streamType == rs2_stream::RS2_STREAM_ANY || depthframe.get_profile().stream_type() == _streamType)) {
```

Comment:
Can we do this outside the worker? So the thread won't even start of not the right frame/stream?